lib/core/viewmodels/schedule_viewmodel.dart

```diff
@@ -170,8 +173,13 @@ class ScheduleViewModel extends FutureViewModel<List<CourseActivity>> {
 
       Future.delayed(const Duration(milliseconds: 700),
           () => FeatureDiscovery.discoverFeatures(context, ids));
-
-      _settingsManager.setString(PreferencesFlag.discoverySchedule, 'true');
     }
   }
+
+  /// Mark the discovery of this view completed
+  Future<bool> discoveryCompleted() async {
+    await _settingsManager.setBool(PreferencesFlag.discoverySchedule, true);
+
+    return true;
+  }
```

Comment:
Isn't this the same behavior as the `recordStepsInSharedPreferences: true` attributes in lib\main.dart? Why was this chosen instead?

Reply:
This is almost the same behaviour. When you log out then log in again we can restart the discovery which avoids the user to go into their settings and clear the storage of the app (most of the users uninstall/reinstall the application to see discovery again)

Reply:
Good seems perfect then!